doc/classes/Thread.xml

```diff
@@ -60,6 +66,17 @@
 				Returns [constant OK] on success, or [constant ERR_CANT_CREATE] on failure.
 			</description>
 		</method>
+		<method name="suspend" qualifiers="static">
+			<return type="Variant" />
+			<param index="0" name="resume_signal" type="Signal" />
+			<param index="1" name="connect_on_main_thread" type="bool" default="true" />
+			<description>
+				Suspend a thread until the signal provided in the [param resume_signal] argument is emitted. As connecting signals between threads is often discouraged (or blocked by thread guards), the [param connect_on_main_thread] is provided to perform the connection to the signal on the main thread.
+				Similarly to script 'await', if the signal provided is emitted with a single argument, then this function will return it. For multiple arguments, an array is returned.
+				As an additional note, if you intend to await on a script function instead of a signal, use the following syntax instead: [code] Thread.suspend( function().completed ) [/code].
+				This function will fail if called from the main thread.
```

Comment:
we could add this codeblock for usage example (not sure about C# usage tho)
```
	[codeblock lang=gdscript]
	# Suspends a thread until a button is pressed.
	if (Thread.is_main_thread()):
		await button.pressed
	else:
		Thread.suspend( button.pressed )
	[/codeblock]
```

Reply:
Added